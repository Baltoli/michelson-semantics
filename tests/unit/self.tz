$Instructions { SELF } ; 
$Stack .List ;
$ParamType int ;
