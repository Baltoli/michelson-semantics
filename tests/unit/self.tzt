code { SELF } ; 
input { } ;
parameter int ;
self "12345" ;
output { Stack_elt (contract int) "12345" } 
