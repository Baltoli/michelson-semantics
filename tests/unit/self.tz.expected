<generatedTop>
  <k>
    .
  </k>
  <stack>
    #Contract ( #Address ( "InvalidMyAddr" ) , int .TypeAnnotationList .FieldAnnotationList )
  </stack>
  <paramtype>
    int .TypeAnnotationList .FieldAnnotationList
  </paramtype>
  <storagetype>
    #NotSet
  </storagetype>
  <balance>
    #Mutez ( 0 )
  </balance>
  <amount>
    #Mutez ( 0 )
  </amount>
  <now>
    #Timestamp ( -1 )
  </now>
  <myaddr>
    #Address ( "InvalidMyAddr" )
  </myaddr>
  <knownaddrs>
    .Map
  </knownaddrs>
  <sourceaddr>
    #Address ( "InvalidSourceAddr" )
  </sourceaddr>
  <senderaddr>
    #Address ( "InvalidSenderAddr" )
  </senderaddr>
  <chainid>
    #ChainId ( 0 )
  </chainid>
</generatedTop>
